pkg/controllers/nodeclass/validation.go

```diff
@@ -40,6 +60,147 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 			fmt.Sprintf("%q tag does not pass tag validation requirements", offendingTag))
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%q tag does not pass tag validation requirements", offendingTag))
 	}
+	// Auth Validation
+	if !nodeClass.StatusConditions().Get(v1.ConditionTypeSecurityGroupsReady).IsTrue() || !nodeClass.StatusConditions().Get(v1.ConditionTypeAMIsReady).IsTrue() || !nodeClass.StatusConditions().Get(v1.ConditionTypeInstanceProfileReady).IsTrue() || !nodeClass.StatusConditions().Get(v1.ConditionTypeSubnetsReady).IsTrue() {
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "DependenciesNotReady", "Waiting for SecurityGroups, AMIs, Subnets and InstanceProfiles to go true")
+
+		return reconcile.Result{}, nil
+	}
+	nodeClaim := &karpv1.NodeClaim{
+		Spec: karpv1.NodeClaimSpec{
+			NodeClassRef: &karpv1.NodeClassReference{
+				Name: nodeClass.ObjectMeta.Name,
+			},
+		},
+	}
+	tags, err := utils.GetTags(nodeClass, nodeClaim, options.FromContext(ctx).ClusterName)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("getting tags, %w", err)
+	}
+
+	createFleetInput := instance.GetCreateFleetInput(nodeClass, string(karpv1.CapacityTypeOnDemand), tags, mockLaunchTemplateConfig())
+	createFleetInput.DryRun = aws.Bool(true)
+
+	if _, err := n.ec2api.CreateFleet(ctx, createFleetInput); awserrors.IgnoreDryRunError(err) != nil {
+		if awserrors.IgnoreUnauthorizedOperationError(err) != nil {
+			// Dry run should only ever return UnauthorizedOperation or DryRunOperation so if we receive any other error
+			// it would be an unexpected state
+			return reconcile.Result{}, fmt.Errorf("unexpected error during CreateFleet validation: %w", err)
+		}
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "CreateFleetAuthCheckFailed", "Controller isn't authorized to call CreateFleet")
+		return reconcile.Result{}, nil
+	}
+
+	createLaunchTemplateInput := launchtemplate.GetCreateLaunchTemplateInput(mockOptions(*nodeClaim, nodeClass, tags), corev1.IPv4Protocol, "")
+	createLaunchTemplateInput.DryRun = aws.Bool(true)
+
+	if _, err := n.ec2api.CreateLaunchTemplate(ctx, createLaunchTemplateInput); awserrors.IgnoreDryRunError(err) != nil {
+		if awserrors.IgnoreUnauthorizedOperationError(err) != nil {
+			// Dry run should only ever return UnauthorizedOperation or DryRunOperation so if we receive any other error
+			// it would be an unexpected state
+			return reconcile.Result{}, fmt.Errorf("unexpected error during CreateLaunchTemplate validation: %w", err)
+		}
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "CreateLaunchTemplateAuthCheckFailed", "Controller isn't authorized to call CreateLaunchTemplate")
+		return reconcile.Result{}, nil
+	}
+
+	// This should never occur as AMIs should already be resolved during the AMI resolution phase
+	if nodeClass.Status.AMIs == nil {
```

Comment:
This being `nil` doesn't guarantee that the length is above 0, just do a length check and you catch both